src/app/3d/qgs3ddebugwidget.cpp

```diff
@@ -0,0 +1,130 @@
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgis.h"
+
+#include "moc_qgs3ddebugwidget.cpp"
+#include "qgs3ddebugwidget.h"
+#include "qgs3dmapcanvas.h"
+#include "qgscameracontroller.h"
+
+Qgs3DDebugWidget::Qgs3DDebugWidget( Qgs3DMapCanvas *canvas, QWidget *parent ) : QWidget( parent ), m3DMapCanvas( canvas )
+{
+  // set up the widget defined in ui file
+  setupUi( this );
+
+  // set up the fixed width of debug widget
+  mCameraInfoGroupBox->setMinimumWidth( mCameraInfoGroupBox->sizeHint().width() );
+  mCameraInfoGroupBox->adjustSize();
+  scrollAreaWidgetContents->setMinimumWidth( scrollAreaWidgetContents->sizeHint().width() );
+  scrollAreaWidgetContents->adjustSize();
+  scrollArea->setMinimumWidth( scrollArea->sizeHint().width() );
+  scrollArea->adjustSize();
+  this->adjustSize();
+
+  // set up the shadow map block
+  mDebugShadowMapCornerComboBox->addItem( tr( "Top Left" ) );
+  mDebugShadowMapCornerComboBox->addItem( tr( "Top Right" ) );
+  mDebugShadowMapCornerComboBox->addItem( tr( "Bottom Left" ) );
+  mDebugShadowMapCornerComboBox->addItem( tr( "Bottom Right" ) );
+  mDebugShadowMapSizeSpinBox->setClearValue( 0.1 );
+
+  // set up the depth map block
+  mDebugDepthMapCornerComboBox->addItem( tr( "Top Left" ) );
+  mDebugDepthMapCornerComboBox->addItem( tr( "Top Right" ) );
+  mDebugDepthMapCornerComboBox->addItem( tr( "Bottom Left" ) );
+  mDebugDepthMapCornerComboBox->addItem( tr( "Bottom Right" ) );
+  mDebugDepthMapSizeSpinBox->setClearValue( 0.1 );
+
+  // set up the camera info block
+  mNearPlane->setRange( std::numeric_limits<float>::lowest(), std::numeric_limits<float>::max() );
+  mFarPlane->setRange( std::numeric_limits<float>::lowest(), std::numeric_limits<float>::max() );
+  mCameraX->setRange( std::numeric_limits<float>::lowest(), std::numeric_limits<float>::max() );
+  mCameraY->setRange( std::numeric_limits<float>::lowest(), std::numeric_limits<float>::max() );
+  mCameraZ->setRange( std::numeric_limits<float>::lowest(), std::numeric_limits<float>::max() );
+  mLookingX->setRange( std::numeric_limits<float>::lowest(), std::numeric_limits<float>::max() );
+  mLookingY->setRange( std::numeric_limits<float>::lowest(), std::numeric_limits<float>::max() );
+  mLookingZ->setRange( std::numeric_limits<float>::lowest(), std::numeric_limits<float>::max() );
+}
+
+/**
+ * Sets up the interactive elements with values from Qgs3DMapSettings
+ */
```

Comment:
Common practice in QGIS code is to have function docstrings in the header... (also for updateFromCamera())